src/components/nested/NestedComponent.js

```diff
@@ -42,6 +42,18 @@ export default class NestedComponent extends BaseComponent {
     return this.components;
   }
 
+  getAllComponents() {
+    return this.getComponents().reduce((components, component) => {
+      let result = component;
+
+      if (component.type === 'form') {
+        result = component.subForm.getAllComponents();
```

Comment:
This will only grab the components at the first level of the nested form. What if the form has multiple levels deep of nested components and you wish to grab the components within them? Will this still work?

Reply:
Also, I don't like have ```component.type === 'form'``` anywhere. This is what we have derived classes for. You should create a method within the form component that does this.

Reply:
@travist It's a recursion there. I've checked nested forms up to three-level depth and everything works correctly.

I have changed a `component.type === 'form'` part as you suggested.